### PR TITLE
Update azdata to 1.25

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/microsoft/azuredatastudio/issues/13592
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/azdata/index.d.ts [hunk 1]
```diff
@@ -14,5267 +13,5272 @@
...
-    export enum ConnectionOptionSpecialType {
-        connectionName = 'connectionName',
-        serverName = 'serverName',
-        databaseName = 'databaseName',
-        authType = 'authType',
-        userName = 'userName',
-        password = 'password',
-        appName = 'appName'
-    }
+	import * as vscode from 'vscode';
```
Comment (danger-public) [inline types/azdata/index.d.ts:21]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

Comment (Charles-Gagnon) [reply types/azdata/index.d.ts]: Thanks, updated

File: types/azdata/index.d.ts [hunk 2]
```diff
@@ -2116,7 +2118,7 @@ declare module 'azdata' {
          * Launches a flyout dialog that will display the information on how to complete device
          * code OAuth login to the user. Only one flyout can be opened at once and each must be closed
          * by calling {@link endAutoOAuthDeviceCode}.
-         * @param providerId    ID of the provider that's requesting the flyout be opened
+         * @param providerId	ID of the provider that's requesting the flyout be opened
```
Comment (danger-public) [inline types/azdata/index.d.ts:2121]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by alanrenmsft — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
Ready to merge

**Pull request merged**